src/Interpreters/ExternalLoader.cpp

```diff
@@ -845,9 +847,10 @@ class ExternalLoader::LoadingDispatcher : private boost::noncopyable
                 if (info.state_id >= min_id)
                     continue;
 
-                all_ready = false;
                 if (info.loading_id < min_id)
                     startLoading(info, forced_to_reload, *min_id);
+
+                all_ready &= (info.state_id >= min_id);
```

Comment:
It's interesting how the loading of models was different from dictionaries... why there was a deadlock for models and everything was good for dictionaries?

Reply:
The problem appears only when `enable_async_loading = true`. In `ExternalDictionariesLoader` it sets to `true` in the constructor, but not in `ExternalModelsLoader`